tasks/service-runner/app/service/service-runner.ts

```diff
@@ -174,6 +175,22 @@ export async function runQueryCmrFromPull(workItem: WorkItemRecord, maxCmrGranul
 
 }
 
+/**
+ * Write logs from the work item execution to s3
+ * @param workItem - the work item that the logs are for
+ * @param logs - logs array from the k8s exec call
+ */
+export async function uploadLogs(workItem: WorkItemRecord, logs: (string | object)[]): Promise<ManagedUpload.SendData> {
+  let fileContent: (string | object)[] = logs;
+  const s3 = objectStoreForProtocol('s3');
+  const logsLocation = getItemLogsLocation(workItem);
+  if (await s3.objectExists(logsLocation)) { // append to existing logs
+    const logsArray = await s3.getObjectJson(logsLocation);
+    fileContent = logsArray.concat(fileContent);
```

Comment:
What do you think about adding a blank line and then 'Start of retry' or something along those lines to separate the messages between retries?

Reply:
Yeah, that sounds like a good idea. I will add that.

Reply:
I just realized with my suggestion you might want to add the line as JSON if the logs are formatted as JSON so that the formatting in a web browser still looks right.

Reply:
No luck with the newline, but I can definitely ensure the format matches whatever is being logged. The message I have so far is:
> 'Start of service execution (retryCount=0, id=0)'